clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/MoveExecutor.scala

```diff
@@ -123,24 +130,29 @@ class MoveExecutor[TI <: TransferIndex](moveCommand: MoveCommand[TI],
   )(implicit ec: ExecutionContext): Future[Option[UpsertId]] = {
 
     val newMetadata = existingMetadata.moveInto(destination, samplePrefix)
-    val preMoveFields = flattenMetadata(existingMetadata)
-    val postMoveFields = flattenMetadata(newMetadata)
+    val preMovePaths = extractPaths(existingMetadata)
+    val postMovePaths = extractPaths(newMetadata)
 
-    val movesToPerform = preMoveFields.flatMap {
-      case (fieldName, path: URI) => {
+    val movesToPerform = preMovePaths.flatMap {
+      case (fieldName, path) => {
         /*
          * Assumptions:
          *   1. If the field exists pre-move, it will still exist post-move
          *   2. If the field is a URI pre-move, it will still be a URI post-move
          */
-        Some(path -> postMoveFields(fieldName).asInstanceOf[URI])
+        Some(path -> postMovePaths(fieldName))
           .filterNot { case (oldPath, newPath) => oldPath.equals(newPath) }
       }
-      case _ => None
     }
 
-    if (movesToPerform.isEmpty) {
-      logger.warn("Nothing to move.")
+    if (preMovePaths.isEmpty) {
+      Future.failed(
+        new IllegalStateException(
+          s"Nothing to move; no files registered to the $name for $prettyKey"
+        )
+      )
+    } else if (movesToPerform.isEmpty) {
+      logger.info(s"Nothing to move; all files already exist at $destination")
```

Comment:
I thought it was OK if all files already exist, in the case where a workflow was re-run?

Reply:
It is OK, the error case is `if (preMovePaths.isEmpty)`. This is still returning `Future.successful(None)` to signal that the move was a no-op, in case anything down the line cares, but I updated the deliver CLP to make it not care.

Reply:
I see. For some reason when I first looked at this I though it was also returning `Future.failed()`, not sure why.